apps/wallet/src/utils/transaction/common.ts

```diff
@@ -404,3 +405,48 @@ export const getNetworkMetadata = (network: Networks) => {
 
 	return { networkIcon, networkName, nativeSymbol };
 };
+
+export const isNativeToken = (
+	network: Networks,
+	tokenSymbol?: string,
+): boolean => {
+	switch (network) {
+		case Networks.solana:
+			return tokenSymbol === 'SOL';
+		case Networks.sui:
+			return tokenSymbol === 'SUI';
+		case Networks.aptos:
+			return tokenSymbol === 'APT';
+		case Networks.tezos:
+			return tokenSymbol === 'XTZ';
```

Comment:
is it safe enough to check via `tokenSymbol`? I think better use `tokenAdress`.

Reply:
It is much better using the `tokenAddress`, the above code is following the checking condition of `constructTransaction` in the same file. I don't think I will resolve this checking approach in this #PR, may I add a `// TODO` and create another ticket to resolve this issue?

Reply:
please use `tokenAddress`, it might cause breaking in the future